From 01aeebfe5ab084a35fa32645d3e2d0c78c2f7dff Mon Sep 17 00:00:00 2001
From: Peter Hoyes <Peter.Hoyes@arm.com>
Date: Tue, 23 Apr 2024 09:10:05 +0100
Subject: [PATCH] armv8: generic_timer: Use event stream for udelay

Upstream-Status: Pending
Signed-off-by: Peter Hoyes <Peter.Hoyes@arm.com>
Signed-off-by: Ross Burton <ross.burton@arm.com>
---
 arch/arm/cpu/armv8/Kconfig | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/cpu/armv8/Kconfig b/arch/arm/cpu/armv8/Kconfig
index dfc4ce851c3..2888075a09c 100644
--- a/arch/arm/cpu/armv8/Kconfig
+++ b/arch/arm/cpu/armv8/Kconfig
@@ -207,6 +207,14 @@ config ARMV8_UDELAY_EVENT_STREAM
 	  delays. This is more efficient than the default polling
 	  implementation.
 
+config ARMV8_UDELAY_EVENT_STREAM
+	bool "Use the event stream for udelay"
+	default y if !ARCH_SOCFPGA
+	help
+	  Use the event stream provided by the AArch64 architectural timer for
+	  delays. This is more efficient than the default polling
+	  implementation.
+
 menuconfig ARMV8_CRYPTO
 	bool "ARM64 Accelerated Cryptographic Algorithms"
 
